Initialize optional clustering directory README (#213)
AlexsLemonade/scpca-downstream-analyses: optional-clustering-analysis/README.md

# Optional Clustering Analysis

> This directory includes a clustering analysis workflow that can be implemented after users have successfully ran the main downstream analysis core workflow as described in this repository's main [README.md](../README.md) file.

Reviewer: In addition to the other comments that Stephanie added, I think before we get too far in the weeds of the actual workflow and setting things up, I think it's important we provide some context and an introduction as to why users would be wanting to use this workflow outside of the clustering that's provided in the main core workflow. I would explain that this can be used to explore different types of clustering and test different parameter values for a given type of clustering which can be helpful in choosing the optimal clustering for each library. I would also emphasize that libraries are unique and the optimal clustering is going to be library dependent.

> There are two main steps of this clustering analysis workflow:

Reviewer 2: I recommend adding a level-2 header here, like "Analysis overview," and then this bit will be included in the TOC. I made another review comment about moving the TOC up as well, see line 19!

1. **Cluster Calculations**: clustering results are calculated for the provided type of graph-based clustering, and range of nearest neighbor values.
Cluster validity and stability results are also calculated.
2. **Cluster Plots**: once clustering results are calculated and stored in the `SingleCellExperiment` object, each of the results are associated with UMAP plots, as well as cluster validity and stability plots to represent the reliability of each set of clustering results.
The plots are displayed in a html report for ease of reference.

> **Note** that the same software requirements for the core workflow are also required for this clustering workflow.

Reviewer 2: I would add a link here to the README section that details requirements. That said, I notice the README in `development` branch still says `4.1` in some places, so we should fix that here too https://github.com/AlexsLemonade/scpca-downstream-analyses/tree/development#3-additional-dependencies

R 4.2 is required for running our pipeline, along with Bioconductor 3.15.

> Package dependencies for the analysis workflows in this repository are managed using [`renv`](https://rstudio.github.io/renv/index.html), and `renv` must be installed locally prior to running the workflow.

Reviewer 2: ```suggestion
Package dependencies for the analysis workflows in this repository are managed using [`renv`](https://rstudio.github.io/renv/index.html), which must be installed locally prior to running the workflow.
```

If you are using conda, dependencies can be installed as [part of the initial setup](../README.md#snakemakeconda-installation).

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

> **Table of Contents**

Reviewer 2: I think this TOC should be higher up in the document (it's hard to even notice the TOC in the rendered README). Also, we can make the introductory stuff a level 2 header which I suggested above.

- [Expected input](#expected-input)
- [Expected output](#expected-output)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Expected input

To run this workflow, you will need to provide:

> 1. The [output `SingleCellExperiment` object](../README.md#expected-output) from the core dowstream analyses workflow.

Reviewer 2: object or RDS file containing the object?

> 1. The [output `SingleCellExperiment` object](../README.md#expected-output) from the core dowstream analyses workflow.

Reviewer 2: see first number in expected output - says you also need a library id!

2. The desired type of graph-based clustering (can be "louvain" or "walktrap"), along with the minimum, maximum, and incremental values that will be used to define the range of nearest neighbor values to be tested.

> For example, if you would like a range of 5:25 to be tested in increments of 5 (5, 10, 15, 20, 25), you will provide `nearest_neighbors_min` = 5, `nearest_neighbors_max` = 25, and `nearest_neighbors_increment` = 5.

Reviewer 2: ```suggestion
For example, if you would like a range of `5:25` to be tested in increments of 5 (as in, `5, 10, 15, 20, 25`), you will provide `nearest_neighbors_min` = 5, `nearest_neighbors_max` = 25, and `nearest_neighbors_increment` = 5.
```

## Expected output

> For each `SingleCellExperiment` RDS file and associated `library_id` used as input, the workflow will return five files:

Reviewer 2: ```suggestion
For each provided `SingleCellExperiment` RDS file and associated `library_id`, the workflow will return five files in the same directory where the inputted RDS file is stored:
```

1. The `SingleCellExperiment` RDS file containing the added clustering results that were calculated in the first step of the worflow.

> 2. The summary html report with the plots representing the clustering results.

Reviewer 2: What's the filename of the report?

> 2. The summary html report with the plots representing the clustering results.

Reviewer 2: ```suggestion
2. The summary html report with plots containing the clustering results.
```

3. A `_clustering_all_validity_stats.tsv` file with all of the calculated cluster validity statistics.
4. A `_clustering_summary_validity_stats.tsv` file with the cluster validity stats averaged across each cluster assignment.
5. A `_clustering_summary_stability_stats.tsv` file with the average adjusted Rand Index (ARI) values across each cluster assignment.

> The latter three files described above will be found in a `clustering_stats` subdirectory stored in the same results directory where the `SingleCellExperiment` RDS file is stored.

Reviewer 2: ```suggestion
The TSV files will be saved in a subdirectory called `clustering_stats/`.
```

I think I took care of the overall directory location with my previous suggestion.

# Optional Clustering Analysis

> This directory includes a clustering analysis workflow that can be implemented after users have successfully ran the main downstream analysis core workflow as described in this repository's main [README.md](../README.md) file.
> 
> The clustering analysis workflow can be used to explore different types of clustering and test a range of parameter values for a given type of clustering.
> Libraries are unique, which means that the optimal clustering is going to be library-dependent.
> Exploring clustering types and parameters can be helpful in choosing the optimal clustering for each library.

Reviewer: ```suggestion
This directory includes a clustering analysis workflow that can help users identify the optimal clustering method and parameters for each library in their dataset. 
Libraries are unique, which means that the optimal clustering is likely to be library-dependent.
The clustering analysis workflow provided here can be used to explore different methods of clustering and test a range of parameter values for a given clustering method in order to identify the optimal clustering for each library.

**The clustering analysis workflow cannot be implemented until after users have successfully run the main downstream analysis core workflow as described in this repository's main [README.md](../README.md) file.**
```
All the content was here I just did some rearranging of the sentence order to first state the overview of how the workflow is helpful and then bolded the note that running the downstream analysis workflow is required to make sure that stands out.

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Analysis overview](#analysis-overview)
- [Expected input](#expected-input)
- [Expected output](#expected-output)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Analysis overview

There are two main steps of this clustering analysis workflow:

1. **Cluster Calculations**: clustering results are calculated for the provided type of graph-based clustering, and range of nearest neighbor values.
Cluster validity and stability results are also calculated.

> 2. **Cluster Plots**: once clustering results are calculated and stored in the `SingleCellExperiment` object, each of the results are associated with UMAP plots, as well as cluster validity and stability plots to represent the reliability of each set of clustering results.

Reviewer: ```suggestion
2. **Cluster Plots**: once clustering results are calculated and stored in the `SingleCellExperiment` object, the results from each of the clustering methods tested are displayed in a UMAP plot.
Additionally, metrics associated with each of the clustering results such as silhouette width, cluster purity, and cluster stability are calculated and plotted.
```
I just tried to clarify this a little bit more since not everyone will understand what you mean by cluster validity and stability plots.

Author: Good point!
